formula/src/main/java/com/instacart/formula/internal/FormulaManagerImpl.kt

```diff
@@ -151,7 +151,23 @@ internal class FormulaManagerImpl<Input, State, Output>(
 
                 result = evaluation
             }
-            return result
+            return result.output
+        } catch (e: Throwable) {
+            if (e is ValidationException) {
+                throw e
+            } else {
+                markAsTerminated()
+                performTerminationSideEffects()
+
+                if (e !is CascadingFormulaException) {
+                    val error = FormulaError.Unhandled(formulaType, e)
```

Comment:
To avoid logging same issue multiple times, we check that it's not `CascadingFormulaException`